packages/components/src/components/post-megadropdown/post-megadropdown.tsx

```diff
@@ -92,34 +100,70 @@ export class PostMegadropdown {
    */
   @Method()
   async show() {
+    this.cancelAllAnimations();
+
     if (PostMegadropdown.activeDropdown && PostMegadropdown.activeDropdown !== this) {
       // Close the previously active dropdown without animation
       PostMegadropdown.activeDropdown.forceClose();
     }
-    this.animationClass = 'slide-in';
 
+    if (!this.isVisible) {
+      this.closeCleanUp();
+    }
+
+    // First set the megadropdown to be visible, then animate
     this.isVisible = true;
     PostMegadropdown.activeDropdown = this;
-    this.postToggleMegadropdown.emit({ isVisible: this.isVisible });
-    if (
-      this.firstFocusableEl &&
-      window.getComputedStyle(this.firstFocusableEl).display !== 'none'
-    ) {
-      this.firstFocusableEl.focus();
+    this.postToggleMegadropdown.emit({ isVisible: true });
+
+    this.currentAnimation = this.createAnimation('in');
+
+    try {
+      await this.currentAnimation.finished;
+
+      // After the megadropdown becomes visible
+      this.currentAnimation = null;
+      this.addListeners();
+
+      if (
+        this.firstFocusableEl &&
+        window.getComputedStyle(this.firstFocusableEl).display !== 'none'
```

Comment:
This check should directly be added to the `getFocusableChildren` function so that `firstFocusableEl` registers the correct element.